liquibase-core/src/main/java/liquibase/parser/core/formattedsql/FormattedSqlChangeLogParser.java

```diff
@@ -422,11 +423,16 @@ public DatabaseChangeLog parse(String physicalChangeLogLocation, ChangeLogParame
                                 Matcher onFailMatcher = onFailPattern.matcher(body);
                                 Matcher onErrorMatcher = onErrorPattern.matcher(body);
                                 Matcher onUpdateSqlMatcher = onUpdateSqlPattern.matcher(body);
+                                Matcher onSqlOutputMatcher = onSqlOutputPattern.matcher(body);
 
                                 PreconditionContainer pc = new PreconditionContainer();
                                 pc.setOnFail(StringUtil.trimToNull(parseString(onFailMatcher)));
                                 pc.setOnError(StringUtil.trimToNull(parseString(onErrorMatcher)));
-                                pc.setOnSqlOutput(StringUtil.trimToNull(parseString(onUpdateSqlMatcher)));
+                                if (onSqlOutputMatcher.matches()) {
```

Comment:
@dyadyushko Can you handle the condition that if changelog contains both onUpdateSql and onSqloutput in which case, we should throw an error, "Please modify the changelog to have preconditions set with either "onUpdateSql" or "onSqlOutput"?